### ISSUE TITLE
[MAYA-122305] Modify -> Center Pivot tool does not preserve the prim's position

### ISSUE THREAD

**Issue by gracekumagai** [Autodesk/maya-usd#2046]
**Describe the bug**
When using the Modify -> Center Pivot tool on a prim with an uncentered pivot does not preserve the prim's position. The same behaviour can be observered:
- when a prim has an uncentered rotate pivot and a rotation applied about that pivot
- when a prim has an uncentered scale pivot and the prim is scaled

**Steps to reproduce**
Steps to reproduce the behavior for the rotate pivot (follow similar steps to repro for the scale pivot):
1. Open a new file in Maya 2022
2. Navigate to Create -> Universal Scene Description (USD) -> Stage with new layer
3. Right click on the proxy shape -> Add new prim -> Capsule
4. Select the Capsule
5. Create xformOp:translate:rotatePivot and xformOp:translateScalePivot by clicking on Modify -> Center Pivot
6. Set xformOp:translate:rotatePivot X to 2 and xformOp:rotateXYZ Z to 180
7. Click Modify -> Center Pivot. Notice that the capsule moves back to the origin.
![center_pivot](https://user-images.githubusercontent.com/12554015/151612585-2a5e94bc-5737-4bd1-9c6c-9e0e0d218d06.gif)

**Expected behavior**
We would expect the same behaviour as native maya when updating pivots where the object does not change position when centering the pivot.

**Specs (if applicable):**
 - maya2022.update3
 - Maya USD dev at 807a118

Pixar Interal JIRA: MAYA-3206

**Comment by gracekumagai** [comment]
Also, when using the Modify -> Center Pivot tool on a prim with xformOp:translate:pivot authored instead of xformOp:translate:rotatePivot and xformOp:translate:scalePivot, it looks like the code is still IP as it prints the following and does not modify the pivot:
```
// Warning: Illegal call to unimplemented UsdSetXformOpUndoableCommandBase::execute() // 
// Warning: Illegal call to unimplemented UsdSetXformOpUndoableCommandBase::redo() // 
```

**Comment by santosg87** [comment]
I am able to reproduce this issue on our side as well. it looks like the xFormOp for piovt adds a transformation to the object space somehow. investigating this a bit deeper

**Comment by santosg87** [comment]
@gracekumagai this is now fixed on Maya 2023.2 and MayaUSD 0.19.0. the fix for this required chanfes on the maya code though, so this will need Maya 2023.2.

marking this closed for the time being. :)

**Comment by gracekumagai** [comment]
Tested in 2023.2 and it's fixed! thank you!
